pkg/manager/diff.go

```diff
@@ -294,8 +294,15 @@ func (kc *kernelContext) Loop() error {
 	if err := kc.serv.Listen(); err != nil {
 		return fmt.Errorf("failed to start rpc server: %w", err)
 	}
-	kc.pool.Loop(kc.ctx)
-	return nil
+	eg, ctx := errgroup.WithContext(kc.ctx)
+	eg.Go(func() error {
+		return kc.serv.Serve(ctx)
+	})
+	eg.Go(func() error {
+		kc.pool.Loop(ctx)
```

Comment:
The way we use context here looks strange. Can we stick to one format?
I mean these two declarations:
1. kernelContext.Loop()
2. kernelContext.pool.Loop(context.Context)

Reply:
`kernelContext` keeps `ctx` as a field, so there's no need to receive it as `Loop()` parameter.

It would have been better not to keep it as a field and only pass it as a parameter, but then we also need to refactor `pkg/rpcserver` to pass it to the `MachineChecked()`.

Reply:
General recommendation is to not store the ctx as a struct field.
But here it is more about the interfaces inconsistency.
The better way to refactor it (assuming we'll agree it makes sense) is to enable containedctx linter and fix the code.
Feel free to ignore it here.